monitoring/mongodb/dashboard.py

```diff
@@ -0,0 +1,1397 @@
+
+    panel = StateTimeline(
+        title=title,
+        description=description,
+        dataSource=DATASOURCE,
+        mappings=mappings,
+        targets=[Target(expr=expr, legendFormat="{{pod}}")],
+        **kwargs,
+    )
+    if width:
+        panel.gridPos = GridPos(h=7, w=width, x=0, y=0)
+    return panel
+
+
+# Overview Panels
+mongodb_services_state = mongodb_stat(
+    "MongoDB services state",
+    orientation="horizontal",
+    targets=[
+        Target(
+            expr=MongoDBMetrics.up(JOBS["configsvr"]),
+            legendFormat="Config Server up:",
+        ),
+        Target(
+            expr=MongoDBMetrics.up(JOBS["mongos"]),
+            legendFormat="MongoS up:",
+        ),
+        Target(
+            expr=MongoDBMetrics.up(JOBS["shards"]),
+            legendFormat="Shard Server up:",
+        ),
+    ],
+)
+
+version_panel = Stat(
+    title="Version",
+    dataSource=DATASOURCE,
+    format="string",
+    colorMode="none",
+    orientation="horizontal",
+    textMode="auto",
+    reduceOptions={"calcs": ["lastNotNull"], "fields": "/^mongodb$/", "values": False},
+    textSize=20,
+    wideLayout=True,
+    targets=[
+        Target(
+            expr=MongoDBMetrics.version_info(),
+            format="table",
+            instant=True,
+            legendFormat="{{mongodb}}",
+        )
+    ],
+)
+
+disk_usage_pie = PieChart(
+    title="Disk usage",
+    dataSource=DATASOURCE,
+    displayLabels=["value", "name", "percent"],
+    legendDisplayMode="hidden",
+    pieType="pie",
+    unit=UNITS.BYTES,
+    targets=[
+        Target(
+            expr=MongoDBMetrics.db_stats("indexSize"),
+            legendFormat="Indexes",
+        ),
+        Target(
+            expr=f'{MongoDBMetrics.db_stats("fsUsedSize")} - {MongoDBMetrics.db_stats("indexSize")}',
+            legendFormat="Documents",
+        ),
+        Target(
+            expr=f'{MongoDBMetrics.db_stats("fsTotalSize")} - {MongoDBMetrics.db_stats("fsUsedSize")}',
+            legendFormat="Free",
+        ),
+    ],
+)
+
+avg_doc_size = mongodb_stat(
+    "Average document size",
+    expr=f'avg(mongodb_dbstats_avgObjSize{{database!~"admin|config|local", rs_state="1", job="{JOBS["shards"]}"}})',
+    unit=UNITS.BYTES,
+    decimals=3,
+    description="Average size of documents in the database",
+)
+
+num_docs_per_shard = mongodb_stat(
+    "Number of documents per shard",
+    expr='mongodb_dbstats_objects{rs_state="1", database!~"admin|config|local", rs_nm=~"$rs_nm"}',
+    textMode="value",
+    noValue="0",
+    colorMode="value",
+    thresholds=[Threshold("blue", 0, 0.0)],
+)
+
+size_collections = mongodb_stat(
+    "Size of Collections",
+    unit=UNITS.BYTES,
+    description="MongoDB stores documents in collections. Collections are analogous to tables in relational databases.",
+    targets=[
+        Target(
+            expr=f'max(mongodb_dbstats_storageSize{{database!~"admin|config|local", job="{JOBS["shards"]}"}})',
```

Comment:
we currently have a single database, so may not be an issue ;

however, since we are filtering on database, I imagine mongo reports separate metrics for each database, right?
And so we need to make the sum over the databases?